tests/DriverManagerTest.php

```diff
@@ -173,15 +177,22 @@ public function testDatabaseUrlDeprecated($url, $expected): void
     }
 
     /**
-     * @param mixed $url
-     * @param mixed $expected
+     * @param array<string, mixed>|string $url
+     * @param array<string, mixed>|false  $expected
      *
      * @dataProvider databaseUrls
      */
     public function testDatabaseUrl($url, $expected): void
     {
         if (is_array($url)) {
+            if (isset($url['driverClass'])) {
+                self::markTestSkipped(
+                    'Legacy test case: Merging driverClass into the parsed parameters has to be done in userland now.',
+                );
+            }
```

Comment:
We currently remove an explicitly configured driver class if the URL has a scheme. That might be unexpected because for all other parameters, it works the other way round. Since with the new parser, the app is in charge of that, we need to skip the corresponding tests.